tower/src/util/mod.rs

```diff
@@ -938,3 +940,32 @@ pub trait ServiceExt<Request>: tower_service::Service<Request> {
 }
 
 impl<T: ?Sized, Request> ServiceExt<Request> for T where T: tower_service::Service<Request> {}
+
+/// Convert an `Option<Layer>` into a [`Layer`].
+///
+/// ```
+/// # use std::time::Duration;
+/// # use tower::Service;
+/// # use tower::builder::ServiceBuilder;
+/// # use tower::util::option_layer;
+/// # #[cfg(feature = "timeout")]
+/// # use tower::timeout::TimeoutLayer;
+/// # #[cfg(all(feature = "timeout", feature = "util"))]
+/// # async fn wrap<S>(svc: S) where S: Service<(), Error = &'static str> + 'static + Send, S::Future: Send {
+/// # let timeout = Some(Duration::new(10, 0));
+/// // Layer to apply a timeout if configured
+/// let maybe_timeout = option_layer(timeout.map(TimeoutLayer::new));
+/// # ServiceBuilder::new()
+/// #     .layer(maybe_timeout)
+/// #     .service(svc);
```

Comment:
TIOLI: it might be worth actually showing this being used:
```suggestion
///  ServiceBuilder::new()
///      .layer(maybe_timeout)
///      .service(svc);
```